src/Estimators/include/BipedalLocomotion/ContactDetectors/SchmittTriggerDetector.h

```diff
@@ -0,0 +1,210 @@
+/**
+ * @file SchmittTriggerDetector.h
+ * @authors Prashanth Ramadoss
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_SCHMITT_TRIGGER_DETECTOR_H
+#define BIPEDAL_LOCOMOTION_SCHMITT_TRIGGER_DETECTOR_H
+
+#include <BipedalLocomotion/ContactDetectors/ContactDetector.h>
+
+#include <iostream>
+#include <unordered_map>
+
+namespace BipedalLocomotion
+{
+namespace Estimators
+{
+
+struct SchmittTriggerParams;
+
+/**
+ * Schmitt Trigger thresholding based contact detector
+ * that maintains and updates the contact states for
+ * a prescribed set of contacts
+ */
+class SchmittTriggerDetector : public ContactDetector
+{
+public:
+    SchmittTriggerDetector();
+    ~SchmittTriggerDetector();
+
+    /**
+     * Set contact force intensity and time stamp for an existing contact
+     * @param[in] contactName Internal state of the estimator
+     * @param[in] time time of measurement
+     * @param[in] force contact force intensity (typically contact normal force)
+     * @return True in case of success, false otherwise.
+     */
+    bool setTimedContactIntensity(const std::string& contactName,
+                                  const double& time,
+                                  const double& force);
+
+    /**
+     * Set contact force intensity and time stamp for existing contacts
+     * @param[in] timedForces container of timed force intesities, pair(first, second): (time, force)
+     * @note any contacts in the input container that does not already exist will be ignored
+     * @return True in case of success, false otherwise.
+     */
+    bool setTimedContactIntensities(const std::unordered_map<std::string, std::pair<double, double>>& timedForces);
```

Comment:
I would suggest to abstract a little bit. Maybe, in some other cases you don't use a force but other parameters. Is there a more generic term? Like ``triggerInput``?

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/142/commits/5a0e97011e1545bd422f72a427513f9e5b32d68a